platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/ExceptionProblemRegistry.java

```diff
@@ -35,14 +40,119 @@
 @ServiceScope(Scope.BuildSession.class)
 public class ExceptionProblemRegistry {
 
-    private final Multimap<Throwable, Problem> problemsForThrowables = Multimaps.synchronizedMultimap(HashMultimap.<Throwable, Problem>create());
-
+    private final Multimap<Throwable, Problem> problemsForThrowables = Multimaps.synchronizedMultimap(MultimapBuilder.linkedHashKeys().linkedHashSetValues().<Throwable, Problem>build());
```

Comment:
❓ Is it a performance problem to use a synchronized multimap here? Can we use some kind of concurrent maps instead?

Reply:
Updating the list doesn't require synchronization if you use `ConcurrentHashMap.compute()`. `ConcurrentHashMap` should take care of that. Or are you referring to another list?